syntax="proto2";
package curve.chunkserver;

enum FileSource {
    FILE_SOURCE_LOCAL = 0;
    FILE_SOURCE_REFERENCE = 1;
}

message LocalFileMeta {
    optional bytes user_meta   = 1;
    optional FileSource source = 2;
    optional string checksum   = 3;
}

// attach pb meta
message CurveSnapshotPbAttachMeta {
    message File {
        required string name = 1;
        optional LocalFileMeta meta = 2;
    };
    repeated File files = 2;
};